code/modules/surgery/bodyparts/species_parts/genetics_bodyparts.dm

```diff
@@ -0,0 +1,216 @@
+		if(bodypart == src)
+			continue
+		if(bodypart.limb_id == BODYPART_ID_RABBIT || bodypart.limb_id == BODYPART_ID_DIGITIGRADE)
+			has_rabbit_leg_still = TRUE
+			break
+	if(!has_rabbit_leg_still)
+		jumping_power.Remove(owner)
+	return ..()
+
+/obj/item/bodypart/head/bunny
+	name = "rabbit head"
+	desc = "Comes with a sniffer for carrots."
+	icon_greyscale = 'icons/mob/human/species/misc/genetics_limbs.dmi'
+	is_dimorphic = TRUE
+	limb_id = BODYPART_ID_RABBIT
+	head_flags = HEAD_HAIR|HEAD_LIPS|HEAD_EYESPRITES|HEAD_EYECOLOR|HEAD_EYEHOLES|HEAD_DEBRAIN
+	var/datum/action/cooldown/spell/olfaction/sniffing_power
+
+/obj/item/bodypart/head/bunny/try_attach_limb(mob/living/carbon/new_head_owner, special)
+	. = ..()
+	if(!.)
+		return
+	var/obj/item/organ/external/snout/bunny/bunny_snout = new
+	bunny_snout.transfer_to_limb(src, new_head_owner)
+
+	sniffing_power = new(src)
+	sniffing_power.background_icon_state = "bg_tech_blue"
+	sniffing_power.base_background_icon_state = sniffing_power.background_icon_state
+	sniffing_power.active_background_icon_state = "[sniffing_power.base_background_icon_state]_active"
+	sniffing_power.overlay_icon_state = "bg_tech_blue_border"
+	sniffing_power.active_overlay_icon_state = null
+	sniffing_power.panel = "Genetic"
+	sniffing_power.Grant(new_head_owner)
+
+/obj/item/bodypart/head/bunny/on_removal()
+	sniffing_power.Remove(owner)
+	return ..()
+
+/obj/item/bodypart/chest/bunny
+	name = "rabbit chest"
+	desc = "Ensures the fluffiest hugs are possible."
+	icon_greyscale = 'icons/mob/human/species/misc/genetics_limbs.dmi'
+	limb_id = BODYPART_ID_RABBIT
+	is_dimorphic = TRUE
+
+/obj/item/bodypart/chest/bunny/try_attach_limb(mob/living/carbon/new_limb_owner, special)
+	. = ..()
+	if(!.)
+		return
+	ADD_TRAIT(new_limb_owner, TRAIT_FRIENDLY, ORGAN_TRAIT)
+
+/obj/item/bodypart/chest/bunny/on_removal()
+	REMOVE_TRAIT(owner, TRAIT_FRIENDLY, ORGAN_TRAIT)
+	return ..()
+
+/obj/item/bodypart/arm/left/bunny
+	name = "rabbit left arm"
+	desc = "Ensures the fluffiest hugs are possible."
+	icon_greyscale = 'icons/mob/human/species/misc/genetics_limbs.dmi'
+	limb_id = BODYPART_ID_RABBIT
+
+/obj/item/bodypart/arm/left/bunny/try_attach_limb(mob/living/carbon/new_limb_owner, special)
+	. = ..()
+	if(!.)
+		return
+	ADD_TRAIT(new_limb_owner, TRAIT_FRIENDLY, ORGAN_TRAIT)
+
+/obj/item/bodypart/arm/left/bunny/on_removal()
+	REMOVE_TRAIT(owner, TRAIT_FRIENDLY, ORGAN_TRAIT)
+	return ..()
+
+/obj/item/bodypart/arm/right/bunny
+	name = "rabbit right arm"
+	desc = "Ensures the fluffiest hugs are possible."
+	icon_greyscale = 'icons/mob/human/species/misc/genetics_limbs.dmi'
+	limb_id = BODYPART_ID_RABBIT
+
+/obj/item/bodypart/arm/right/bunny/try_attach_limb(mob/living/carbon/new_limb_owner, special)
+	. = ..()
+	if(!.)
+		return
+	ADD_TRAIT(new_limb_owner, TRAIT_FRIENDLY, ORGAN_TRAIT)
+
+/obj/item/bodypart/arm/right/bunny/on_removal()
+	REMOVE_TRAIT(owner, TRAIT_FRIENDLY, ORGAN_TRAIT)
+	return ..()
```

Comment:
```suggestion
	icon_greyscale = 'icons/mob/human/species/misc/genetics_limbs.dmi'
	limb_id = BODYPART_ID_RABBIT
	bodypart_traits = list(TRAIT_FRIENDLY)
```
You should use the `bodypart_traits` list rather than handling this manually, otherwise currently they will override each-other.